src/hooks/useDetectOs.ts

```diff
@@ -4,11 +4,28 @@ import downloadUrlByOS from '../util/downloadUrlByOS';
 
 export const useDetectOs = () => {
   const [userOS, setUserOS] = useState<UserOS>(UserOS.UNKNOWN);
+  const [bitness, setBitness] = useState<string>('');
 
-  useEffect(() => setUserOS(detectOS), []);
+  useEffect(() => {
+    setUserOS(detectOS);
+
+    // This is necessary to detect Windows 11 on Edge.
+    // [MDN](https://developer.mozilla.org/en-US/docs/Web/API/NavigatorUAData/getHighEntropyValues)
+    // [MSFT](https://learn.microsoft.com/en-us/microsoft-edge/web-platform/how-to-detect-win11)
+    // @ts-expect-error no types for "userAgentData" because this API is experimental
+    navigator?.userAgentData
+      ?.getHighEntropyValues(['bitness'])
```

Comment:
Nit, instead of using optional chaining rather:

```jsx
if (navigator.userAgentData) { ... }
```

Even more, because I'm not sure the optional chaining here would protect you from the `.then` usages... It might give something like "cannot call function then from undefined"